hotel-revisited.md

```diff
@@ -0,0 +1,131 @@
+  def initialize(unit_price, quantity)
+    @unit_price = unit_price
+    @quantity = quantity
+  end
+end
+
+class ShoppingCart
+  attr_accessor :entries
+  def initialize
+    @entries = []
+  end
+end
+
+class Order
+  SALES_TAX = 0.07
+  def initialize
+    @cart = ShoppingCart.new
+  end
+
+  def total_price
+    sum = 0
+    @cart.entries.each do |entry|
+      sum += entry.unit_price * entry.quantity
+    end
+    return sum + sum * SALES_TAX
+  end
+end
+```
+
+#### Implementation B
+
+```ruby
+class CartEntry
+  def initialize(unit_price, quantity)
+    @unit_price = unit_price
+    @quantity = quantity
+  end
+
+  def price
+    return @unit_price * @quantity
+  end
+end
+
+class ShoppingCart
+  def initialize
+    @entries = []
+  end
+
+  def price
+    sum = 0
+    @entries.each do |entry|
+      sum += entry.price
+    end
+    return sum
+  end
+end
+
+class Order
+  SALES_TAX = 0.07
+  def initialize
+    @cart = ShoppingCart.new
+  end
+
+  def total_price
+    subtotal = @cart.price
+    return subtotal + subtotal * SALES_TAX
+  end
+end
+```
+
+#### Prompts
+
+Once you have read through the above code, add a file to your Hotel project called `design-activity.md`. This file will be submitted with the rest of this exercise. In that file, please respond to the following prompts:
+
+- What classes does each implementation include? Are the lists the same?
+- Write down a sentence to describe each class.
+- How do the classes relate to each other? It might be helpful to draw a diagram on a whiteboard or piece of paper.
+- What **data** does each class store? Are they different for the two implementations?
+- What **methods** does each class have? Are they different for the two implementations?
+- Consider the `Order#total_price` method. In each implementation:
+    - Is logic to compute the price delegated to "lower level" classes like `ShoppingCart` and `CartEntry`, or is it retained in `Order`?
+    - Does `total_price` directly manipulate the instance variables of other classes?
+- Which implementation better adheres to the single responsibility principle?
+
+Once you've responded to the prompts, `git add design-activity.md` and `git commit`!
+
+## Revisiting Hotel
+
+Now that we've got you thinking about design, spend some time to revisit the code you wrote for the Hotel project. For each class in your program, ask yourself the following questions:
+- What is this class's responsibility?
+    - You should be able to describe it in a single sentence.
+- Is this class responsible for exactly one thing?
+- Does this class take on any responsibility that should be delegated to "lower level" classes?
+- Is there code in other classes that directly manipulates this class's instance variables?
+
+### Activity
+
+Identify one place in your Hotel project where you believe your design could be improved. Describe in `design-activity.md` what changes you would need to make to actualize that improvement, and why.
```

Comment:
Maybe some explicit prompting to look for areas where a class takes on multiple roles, or directly modifies the attributes of another class.